pkg-manager/core/src/install/index.ts

```diff
@@ -379,6 +381,7 @@ export async function mutateModules (
         throw new LockfileConfigMismatchError(outdatedLockfileSettingName!)
       }
     }
+    const _isWantedDepPrefSame = isWantedDepPrefSame.bind(null, ctx.wantedLockfile.catalogs, opts.catalogs)
```

Comment:
I usually prefer explicitly curried functions like this:

```ts
function createIsWantedDepPrefSameFn(
  prevCatalogs: CatalogSnapshots | undefined,
  catalogsConfig: Catalogs | undefined
) {
  return function isWantedDepPrefSame(/* ... */) { /* ... */ }
}
```

The code block here would then be:

```ts
const isWantedDepPrefSame = createIsWantedDepPrefSameFn(ctx.wantedLockfile.catalogs, opts.catalogs)
```

I like that a bit more since it avoids the underscored function, the `null` argument, and makes the function currying API more clear. But I know the pnpm codebase usually prefers `.bind()`.

Reply:
It seems to me like you need to pass catalogs to forgetResolutionsOfPrevWantedDeps instead of passing the function to it.

Reply:
The idea is to have each function's parameters match what they should concerned with:

- `forgetResolutionsOfPrevWantedDeps` is concerned with which previously wanted deps are the same.
- `isWantedDepPrefSame` needs to know about catalogs to answer the question it's named after.

The current design matches the ideals of functional programming. Instead of a large function that's concerned with many concepts, we pass around smaller functions that are more limited in scope and easier to understand.